app/pages/_app.js

```diff
@@ -18,7 +18,7 @@ const PageLayout = ({ children }) => children;
 class App extends NextApp {
   componentDidCatch(error, errorInfo) {
     console.log(error, errorInfo);
-    // TODO(naman): Add ErrorHandler here
+    pageProps = { ...pageProps, hasError: true };
```

Comment:
You need to store the hasError state in the App Component only to wrap the error. The error got propagated to the App component from the children and the current way you are still rendering the children. The app component has to stop rendering the children and render the ErrorLayout.

Reply:
okay i thought that createPage also has Error handler so it will decide for each page whether to render default component or error component.i am making the changes.

Reply:
createPage does and it is based on the error variable, no need to accept hasError here.